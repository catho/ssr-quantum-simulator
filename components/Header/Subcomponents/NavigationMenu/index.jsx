import SandwichButton from '../SandwichButton';

const NavigationMenu = () => (
  <>
    <nav
      aria-label="Menu Principal"
      id="visibleDesktopMenu"
      className="VisibleMenu FlexEndPositioning"
    >
      <ul role="menu" className="NavList">
        <li role="menuitem" className="DisplayNoneOnSmallScreen">
          <a href="/cathocomponents">Catho Components</a>
        </li>
        <li role="menuitem" className="DisplayNoneOnSmallScreen">
          <a href="/quantum">Quantum</a>
        </li>
      </ul>
    </nav>
    <SandwichButton />
    <nav
      aria-label="Menu de opções"
      id="hiddenDesktopMenu"
      className="HiddenMenu"
    >
      <ul role="menu">
        <li role="menuitem" className="HiddenMenuNavigationItem">
          <a href="/cathocomponents">Catho Components</a>
        </li>
        <li role="menuitem" className="HiddenMenuNavigationItem">
          <a href="/quantum">Quantum</a>
        </li>
        <li role="menuitem">
          <a href="/render">Simulates renders</a>
        </li>
        <li role="menuitem">
          <a href="https://github.com/catho">Repositories</a>
        </li>
      </ul>
    </nav>
  </>
);

export default NavigationMenu;
