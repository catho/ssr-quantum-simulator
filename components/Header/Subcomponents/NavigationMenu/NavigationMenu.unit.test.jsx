import { render, screen } from '@testing-library/react';
import NavigationMenu from '.';

test('should match the snapshot', () => {
  const { container } = render(<NavigationMenu />);
  expect(container).toMatchSnapshot();
});

it('should render the links and sandwitch button', () => {
  render(<NavigationMenu />);

  const cathoComponentsLinks = screen.getAllByText(/Catho Components/i);
  const cathoComponentsVisibleMenuLink = cathoComponentsLinks[0];
  const cathoComponentsMobileMenuLink = cathoComponentsLinks[1];

  const quantumLinks = screen.getAllByText('Quantum');
  const quantumVisibleMenuLink = quantumLinks[0];
  const quantumMobileMenuLink = quantumLinks[1];

  const sandwitchBtn = screen.getByTestId('sandwichButton');
  const simulatesRendersMenuLink = screen.getByText('Simulates renders');
  const repositoriesMenuLink = screen.getByText('Repositories');

  expect(cathoComponentsVisibleMenuLink).toBeInTheDocument();
  expect(quantumVisibleMenuLink).toBeInTheDocument();
  expect(sandwitchBtn).toBeInTheDocument();
  expect(simulatesRendersMenuLink).toBeInTheDocument();
  expect(repositoriesMenuLink).toBeInTheDocument();
  expect(cathoComponentsMobileMenuLink).toBeInTheDocument();
  expect(quantumMobileMenuLink).toBeInTheDocument();
});
