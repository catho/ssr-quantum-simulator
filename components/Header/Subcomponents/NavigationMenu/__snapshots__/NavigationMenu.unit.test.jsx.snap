// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <nav
    aria-label="Menu Principal"
    class="VisibleMenu FlexEndPositioning"
    id="visibleDesktopMenu"
  >
    <ul
      class="NavList"
      role="menu"
    >
      <li
        class="DisplayNoneOnSmallScreen"
        role="menuitem"
      >
        <a
          href="/cathocomponents"
        >
          Catho Components
        </a>
      </li>
      <li
        class="DisplayNoneOnSmallScreen"
        role="menuitem"
      >
        <a
          href="/quantum"
        >
          Quantum
        </a>
      </li>
    </ul>
  </nav>
  <input
    aria-controls="hiddenDesktopMenu"
    aria-expanded="false"
    aria-label="Menu"
    id="menuSandwichButton"
    type="checkbox"
  />
  <label
    class="SandwichButton"
    data-testid="sandwichButton"
    for="menuSandwichButton"
  >
    ☰
  </label>
  <nav
    aria-label="Menu de opções"
    class="HiddenMenu"
    id="hiddenDesktopMenu"
  >
    <ul
      role="menu"
    >
      <li
        class="HiddenMenuNavigationItem"
        role="menuitem"
      >
        <a
          href="/cathocomponents"
        >
          Catho Components
        </a>
      </li>
      <li
        class="HiddenMenuNavigationItem"
        role="menuitem"
      >
        <a
          href="/quantum"
        >
          Quantum
        </a>
      </li>
      <li
        role="menuitem"
      >
        <a
          href="/render"
        >
          Simulates renders
        </a>
      </li>
      <li
        role="menuitem"
      >
        <a
          href="https://github.com/catho"
        >
          Repositories
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
